go/worker/common/p2p/dispatch.go

```diff
@@ -153,13 +156,15 @@ func (h *topicHandler) dispatchMessage(peerID core.PeerID, m *queuedMsg, isIniti
 	// Perhaps this should reject the message, but it is possible that
 	// the local node is just behind.  This does result in stale messages
 	// getting retried though.
-	if err = h.handler.AuthenticatePeer(m.from, m.msg); err != nil {
-		return err
-	}
+	for _, handler := range h.handlers {
+		if err = handler.AuthenticatePeer(m.from, m.msg); err != nil {
```

Comment:
This should probably be split into two traversals, first we authenticate the peer using all handlers and stop if any fails. Otherwise partial dispatches could be possible.

Then we dispatch the message to all handlers. The question is should we always dispatch to all first before returning an error in case one of the handlers failed or not?